Failed tests upon clean checkout

**Describe the bug**
Post checkout, I built using Ninja and passed `--test` flag. This resulted in the following failed tests:

```
Failed Tests (4):
  Swift(macosx-arm64) :: Interop/Cxx/class/inheritance/fields.swift
  Swift(macosx-arm64) :: Interop/Cxx/class/inheritance/functions.swift
  Swift(macosx-arm64) :: Interop/Cxx/foreign-reference/pod.swift
  Swift(macosx-arm64) :: Interop/Cxx/operators/member-inline.swift
```

**Steps To Reproduce**
1. Follow the steps described in [Cloning the project](https://github.com/apple/swift/blob/main/docs/HowToGuides/GettingStarted.md#cloning-the-project).
2. At the time of raising this issue, the `main` branch is pointing at commit `15e17b2`.
3. Build via Ninja (notice the `--test` flag)
```
utils/build-script --skip-build-benchmarks \
  --skip-ios --skip-watchos --skip-tvos --swift-darwin-supported-archs "$(uname -m)" \
  --sccache --release-debuginfo --swift-disable-dead-stripping --test
```

**Expected behavior**
All tests passes.

**Environment (please fill out the following information)**
 - OS: macOS 12.3
 - Xcode 13.4

**Additional context**
Full build logs

```
Failed Tests (4):
  Swift(macosx-arm64) :: Interop/Cxx/class/inheritance/fields.swift
  Swift(macosx-arm64) :: Interop/Cxx/class/inheritance/functions.swift
  Swift(macosx-arm64) :: Interop/Cxx/foreign-reference/pod.swift
  Swift(macosx-arm64) :: Interop/Cxx/operators/member-inline.swift


Testing Time: 578.24s
  Unsupported      :  486
  Passed           : 8412
  Expectedly Failed:   24
  Failed           :    4

4 warning(s) in tests
FAILED: test/CMakeFiles/check-swift-macosx-arm64 /Users/mennabah/projects/os/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test/CMakeFiles/check-swift-macosx-arm64
cd /Users/mennabah/projects/os/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test && /opt/homebrew/Cellar/cmake/3.24.1/bin/cmake -E remove_directory /Users/mennabah/projects/os/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/./swift-test-results/arm64-apple-macosx10.9 && /opt/homebrew/Cellar/cmake/3.24.1/bin/cmake -E make_directory /Users/mennabah/projects/os/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/./swift-test-results/arm64-apple-macosx10.9 && /opt/homebrew/Cellar/cmake/3.24.1/bin/cmake -E env /Applications/Xcode.app/Contents/Developer/usr/bin/python3 /Users/mennabah/projects/os/swift-project/llvm-project/llvm/utils/lit/lit.py --param bootstrapping_mode=BOOTSTRAPPING-WITH-HOSTLIBS -sv -j 10 --xunit-xml-output=/Users/mennabah/projects/os/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/./swift-test-results/arm64-apple-macosx10.9/lit-tests.xml --param differentiable_programming --param concurrency --param distributed --param string_processing --param threading=darwin --param swift_test_subset=primary --param swift_test_mode=optimize_none /Users/mennabah/projects/os/swift-project/build/Ninja-RelWithDebInfoAssert/swift-macosx-arm64/test-macosx-arm64
ninja: build stopped: subcommand failed.
ERROR: command terminated with a non-zero exit status 1, aborting
```

--- a/test/CMakeLists.txt
+++ b/test/CMakeLists.txt
@@ -43,6 +43,10 @@ function(get_test_dependencies SDK result_var_name)
     list(APPEND deps SwiftUnitTests)
   endif()
 
+  if(SWIFT_BUILD_SDK_OVERLAY OR SWIFT_BUILD_TEST_SUPPORT_MODULES)
+    list(APPEND deps sdk-overlay)
+  endif()
+
   set(deps_binaries)
 
   if (SWIFT_INCLUDE_TOOLS)

--- a/test/Interop/Cxx/class/constructors-copy-irgen.swift
+++ b/test/Interop/Cxx/class/constructors-copy-irgen.swift
@@ -4,6 +4,9 @@
 // RUN: %swift -module-name Swift -target armv7-none-linux-androideabi -dump-clang-diagnostics -I %S/Inputs -enable-experimental-cxx-interop -emit-ir %s -parse-stdlib -parse-as-library -disable-legacy-type-info | %FileCheck %s -check-prefix=ITANIUM_ARM
 // RUN: %swift -module-name Swift -target x86_64-unknown-windows-msvc -dump-clang-diagnostics -I %S/Inputs -enable-experimental-cxx-interop -emit-ir %s -parse-stdlib -parse-as-library -disable-legacy-type-info | %FileCheck %s -check-prefix=MICROSOFT_X64
 
+// REQUIRES: CODEGENERATOR=X86
+// REQUIRES: CODEGENERATOR=ARM
+
 import Constructors
 import TypeClassification
 

--- a/test/Interop/Cxx/class/constructors-irgen.swift
+++ b/test/Interop/Cxx/class/constructors-irgen.swift
@@ -4,6 +4,9 @@
 // RUN: %swift -module-name Swift -target armv7-unknown-linux-androideabi -dump-clang-diagnostics -I %S/Inputs -enable-experimental-cxx-interop -emit-ir %s -parse-stdlib -parse-as-library -disable-legacy-type-info | %FileCheck %s -check-prefix=ITANIUM_ARM
 // RUN: %swift -module-name Swift -target x86_64-unknown-windows-msvc -dump-clang-diagnostics -I %S/Inputs -enable-experimental-cxx-interop -emit-ir %s -parse-stdlib -parse-as-library -disable-legacy-type-info | %FileCheck %s -check-prefix=MICROSOFT_X64
 
+// REQUIRES: CODEGENERATOR=X86
+// REQUIRES: CODEGENERATOR=ARM
+
 import Constructors
 import TypeClassification
 

